frontend/ui/widget/inputdialog.lua

```diff
@@ -855,7 +868,7 @@ function InputDialog:_addScrollButtons(nav_bar)
     end
     if nav_bar then -- Add the Home & End buttons
         -- Also add Keyboard hide/show button if we can
-        if self.fullscreen and not self.readonly then
+        if self.fullscreen and not self.readonly and Device:isTouchDevice() and not Device:hasKeyboard() then
```

Comment:
```suggestion
        if self.fullscreen and not self.readonly and not (Device:hasKeyboard() and G_reader_settings:isFalse("virtual_keyboard_enabled")) then
```
@poire-z  would only remove it to people who hide VK, still would not work on NT with no `hasKeyboard` though

Reply:
I guess your point was to have the VK available now for people who hasKeyboard (because they may find in there chars not available on the physical keyboard).
So I guess you could juste remove `and Device:isTouchDevice() and not Device:hasKeyboard()` - and don't make everything too complex with these device:hasSomething().

Either they see the nav_bar fully (and they can use it with touch or dpad or something else), or they don't see it at all.

Reply:
in lame christian terms, are you saying: keep it as `if self.fullscreen and not self.readonly` or as in the suggestion here?

Reply:
> keep it as `if self.fullscreen and not self.readonly`

I meant this ^